core/src/main/java/hudson/util/io/ReopenableFileOutputStream.java

```diff
@@ -37,8 +37,9 @@
  * and then keep writing.
  *
  * @author Kohsuke Kawaguchi
+ * @deprecated due to risk for file leak
```

Comment:
This isn't an alternative, but a subset of what ReopenableFileOutputStream intended to do.